tests/public_api_tests/test_backwards_compatible.py

```diff
@@ -0,0 +1,52 @@
+# Copyright (c) 2013 Shotgun Software Inc.
+#
+# CONFIDENTIAL AND PROPRIETARY
+#
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+from unittest2 import TestCase
+
+import sgtk
+
+
+class BackwardsCompatibilityTests(TestCase):
+
+    def test_class_availability(self):
+        """
+        Ensures the API is backwards compatible.
+        """
+
+        # Each test should be interpreter as
+        #
+        # self.assertEqual(new_location, old_location)
+
+        # Utils backwards compatibility
+        self.assertEqual(
+            sgtk.authentication.CoreDefaultsManager,
+            sgtk.util.CoreDefaultsManager
+        )
+
+        # Descriptor backwards compatibility
+        self.assertEqual(
+            sgtk.descriptor.TankInvalidAppStoreCredentialsError,
```

Comment:
I think we should decentralize instead of building a monolith. We want to keep `descriptor` fully decoupled from `authentication` etc. So no, it doesn't make sense to collect things in one place. i want to run the descriptor tests and know that all descriptor related tests have executed :) imagine a future were we separate the descriptor API from the rest of core. `pip install tk-core-descriptor` :)